### PR TITLE
Prevent switching to fullscreen, if window has sf::Style::None and desktop resolution on windows.

### PR DESCRIPTION
**Pull request by Foaly**
This is a fix for #1284. It would be great if you could test it @naezith @Jaez @marukrap

### CODE REVIEW

File: src/SFML/Window/Win32/WindowImplWin32.cpp [hunk 1]
```diff
@@ -193,11 +193,7 @@ m_cursorGrabbed   (m_fullscreen)
 
     // Choose the window style according to the Style parameter
     DWORD win32Style = WS_VISIBLE;
-    if (style == Style::None)
-    {
-        win32Style |= WS_POPUP;
-    }
-    else
+    if (style != Style::None)
```
Comment (LaurentGomila) [inline src/SFML/Window/Win32/WindowImplWin32.cpp]: No need to test explicitly against Style::None, in this case none of the tests below will succeed and the `win32Style` variable will correctly stay at its initial value.

File: src/SFML/Window/Win32/WindowImplWin32.cpp [hunk 2]
```diff
@@ -216,6 +212,12 @@ m_cursorGrabbed   (m_fullscreen)
     // Create the window
     m_handle = CreateWindowW(className, title.toWideString().c_str(), win32Style, left, top, width, height, NULL, NULL, GetModuleHandle(NULL), this);
 
+    // Remove all decorations if requested
+    if (style == Style::None)
+    {
+        SetWindowLong(m_handle, GWL_STYLE, 0);
```
Comment (LaurentGomila) [inline src/SFML/Window/Win32/WindowImplWin32.cpp]: How did you find this solution? Is it documented?

Comment (MarioLiebisch) [reply src/SFML/Window/Win32/WindowImplWin32.cpp]: This basically sets the window to [`WS_OVERLAPPED`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms632600(v=vs.85).aspx):

> The window is an overlapped window. An overlapped window has a title bar and a border. Same as the **WS_TILED** style.

This can't modify all styles after the window is created, so we're potentially just reverting one of the flags rather than everything and this works just because...

Comment (LaurentGomila) [reply src/SFML/Window/Win32/WindowImplWin32.cpp]: How is it different from the style argument that we pass to `CreateWindowW`? Should be the same thing, and we already pass 0 to `CreateWindowW` (we just add `WS_VISIBLE` which has nothing to do with decorations).

Comment (MarioLiebisch) [reply src/SFML/Window/Win32/WindowImplWin32.cpp]: At window creation all components of `WS_OVERLAPPED` can be applied, so there might be happening more than when changing the flags only.

Comment (LaurentGomila) [reply src/SFML/Window/Win32/WindowImplWin32.cpp]: The MSDN documentation of the `dwStyle` argument of `CreateWindow` and of the `GWL_STYLE` option of `SetWindowLong` both directly point to the window styles page, so (if the doc is correct) that should be the exact same thing.

**Comment by Foaly** [comment]
Hmm no it does not seem to be documented. I took it from [here](https://stackoverflow.com/questions/7442939/opening-a-window-that-has-no-title-bar-with-win32) and assumed that 0 meant no state. I didn't double check with the windows documentation. This fixes the bug though. Tested on windows 8.1 by the way.

**Comment by LaurentGomila** [comment]
If we can't find a documented source, I'm afraid we'll need more tests before we can validate it.

**Comment by hobby8** [comment]
I suggest to examine with [Spy++](https://docs.microsoft.com/en-us/visualstudio/debugger/introducing-spy-increment) the window styles of a few applications that get this right on the problem OS.
If doing the same does not help, you may want to try a tool such as [API Monitor](http://www.rohitab.com/apimonitor) (enable 'Windows Application UI Development/Windows and Messages/Windows' and maybe also 'Graphics/Legacy Graphics').

**Comment by oomek** [comment]
I'm actually using sf::Style::None when creating a fullscreen window on Windows 10. I set the resolution to match the screen resolution. This way I get all the benefits of fullscreen (double/triple buffering, fullscreen optimizations) but with much faster and seamless window creation. It seems Windows 10 detects it straight away I want a borderless fullscreen window. Wouldn't this commit break it? If so, wouldn't it be a good idea to add another flag sf::Style:Borderless and keep the current sf::Style::None behaviour?
